fmriprep/cli/run.py

```diff
@@ -71,13 +73,15 @@ def main():
     config.load(config_file)
 
     if config.execution.reports_only:
+        ping_migas(status="success" if retcode == 0 else "error")
         sys.exit(int(retcode > 0))
 
     if fmriprep_wf and config.execution.write_graph:
         fmriprep_wf.write_graph(graph2use="colored", format="svg", simple_form=True)
 
     retcode = retcode or (fmriprep_wf is None) * EX_SOFTWARE
     if retcode != 0:
+        ping_migas(status="error")
```

Comment:
What about replacing all these `ping_migas` with status with an exit hook? - https://docs.python.org/3/library/atexit.html

Since the hook does not allow to access the exit code, we probably want to:
1. Add a new `status` or `exitcode` config setting (perhaps under the `execution` section). Initialize it at -1.
2. Define some codes: e.g., 1001 (I haven't really thought this through, this might not be a good code) is set right before building the workflow, then 1002 is set after it is built and before calling `run()` and finally 0 is set when it is complete and successful. An error in the pipeline overrides the exitcode too.
3. The `atexit()` hook will not send a ping if the code is still -1, and otherwise, submit whatever code it has.